### ISSUE TITLE
CshtmlComponentInitial specify location of html output

### ISSUE THREAD

**Issue by nyash** [Acmion/CshtmlComponent#1]
Hi,

First thanks for the library seems like a great attempt at fixing some of shortcommings of TagHelpers/ViewComponents for those that don't want/can't go full out Blazor.

One thing I had noticed is that the content specified in the *CshtmlComponentInitial* tag is rendered as is wherever the first component using it is located.

For example I would very much like to output such content in the *head* tag or maybe before *body* end.

I think it would be very useful to be able to specify (maybe via another custom tag and/or custom attribute on script/style/link elements within CsComponentInitial?) where the output of CshtmlComponentInitial should be rendered.
 
Thanks!

**Comment by Acmion** [comment]
Thanks for the comment!

I'd like to keep the current behavior of `<CshtmlInitial>`. 

That said, I could implement the following components: 
`<CshtmlPostHead>`  
which would then append all of its content to the end of the `<head>` tag. 

`<CshtmlPostBody>`  
which would then append all of its content to the end of the `<body>` tag. 

However, they would still require the `Context` attribute, just like `<CshtmlInitial>` (due to some limitations in ASP.NET Core that I haven't found any solutions to). In other words, they would have to be rendered as:
```
<CshtmlPostBody Context="@Model">
    HTML here
</CshtmlPostBody>
```

Would this be good enough?

And how important do you think that the order would be? For example:  
Page 1:
Component A  
Component B  

Page 2:
Component B  
Component A  

The post body content would be in a different order for both pages. Is this an important issue?

**Comment by Acmion** [comment]
This is now implemented in V3.0.0.
